engine/runner/src/main/java/org/enso/runner/EnsoLibraryFeature.java

```diff
@@ -0,0 +1,93 @@
+package org.enso.runner;
+
+import static scala.jdk.javaapi.CollectionConverters.asJava;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashSet;
+import java.util.TreeSet;
+import org.enso.compiler.core.ir.module.scope.imports.Polyglot;
+import org.enso.pkg.PackageManager$;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+public final class EnsoLibraryFeature implements Feature {
+  @Override
+  public void beforeAnalysis(BeforeAnalysisAccess access) {
+    var libs = new LinkedHashSet<Path>();
+    for (var p : access.getApplicationClassPath()) {
+      var p1 = p.getParent();
+      if (p1 != null && p1.getFileName().toString().equals("java")) {
+        var p2 = p1.getParent();
+        if (p2 != null
+            && p2.getFileName().toString().equals("polyglot")
+            && p2.getParent() != null) {
+          libs.add(p2.getParent());
+        }
+      }
+    }
+
+    /*
+      To run Standard.Test one shall analyze its polyglot/java files. But there are none
```

Comment:
The main purpose of this PR is to make sure we **can run (some subset) of `Base_Tests`**. For that we need to include all the `polyglot java import` statements we find even in `Standard.Test` and even in `test/Base_Tests`.

The other _ultimate purpose_ of these changes is to build **production image**. Such an image shouldn't have the `test/Base_Tests` imports. 

To achieve both goals we (currently) would need to build twice and have a property to tell us to include imports from some special directory - e.g. `test/Base_Tests`.

Reply:
> would need to build twice

I'm slightly anxious about building a different set for tests and production.

You have shown us that for this to work we need to add imports of some of the used classes, to make them available at runtime.

What if we have some class `A` used as part of implementation of a feature in the library, but never imported there. If the class `A` is then imported in Tests, our tests of that feature will succeed because the test build will include the `A` access. But then the same feature may stop working in a production build because that one will not include it. What can we do to prevent such scenarios from happening in a dual-build setup?

Reply:
That's how testing `native-image` tests work. A new image with tests classes has to be built - see [Junit support in NI](https://github.com/graalvm/native-build-tools/blob/master/common/junit-platform-native/README.md).

The only way out of this is to avoid usage of special test classes and test just pure Enso code. Shall I report an issue to remove tests that need 
```
org.enso.base_test_helpers.CallbackHelper
org.enso.base_test_helpers.Number_Utils
org.enso.base_test_helpers.IntHolder
org.enso.base_test_helpers.IntHolderEquals
org.enso.base_test_helpers.CallbackHelper
org.enso.base_test_helpers.IntHolder
org.enso.base_test_helpers.FileSystemHelper
```
from `Base_Tests` and move them into different project? Then we wouldn't need to rebuild the executable and could use just the official one.

Reply:
Hmm, I see.

Well the classes in `org.enso.base_test_helpers` seemed to be harmless because we don't depend on them in the implementation. I was worried we can miss a class actually used in implementation because of the above. Not sure what is the right way out of this though, as you point out it is non-trivial to distinguish test classes from non-test in an 'automatic' way.

Reply:
@radeusgd what do you want to do with [this import](https://github.com/enso-org/enso/blob/3b59b4065772fe9255daad73b3adf57380d30377/test/Base_Tests/src/Data/Ordering/Comparator_Spec.enso#L6)? I can image:
- we remove the import of `org.enso.base.CompareException` and `org.enso.base.ObjectComparator`, remove the related tests and test just the Enso API of `Ordering` - then we can use **production executable** to run the test
- we modify code in `Ordering.enso` to also perform the import - just [like here](https://github.com/enso-org/enso/pull/9997/files#diff-98679583c906bee2f7c467b9489112f06bdd308b74044d7d28f322495ff5d559R16) - then we can use **production executable** to run the test - btw. that the _current state_ of the PR
- we play some tricks to _rebuild the executable image_ with these additional imports - then we'll be _testing different bits_ than we have in production

What shall I do (in this PR)? Keeping current [status quo](https://github.com/enso-org/enso/pull/9997/files#diff-98679583c906bee2f7c467b9489112f06bdd308b74044d7d28f322495ff5d559R16) seems simplest.

Reply:
I think the current state is fine. 

Note that `CompareException` is already imported in `Standard.Base` (`Statistics.enso`).

Incidentally, the `ObjectComparator` is imported in `Standard.Table`, so maybe the best option would be to move the `Comparator_Spec` to `Table_Tests` instead? That would solve the import issue, and actually this part of comparator functionality is used more in Table IIRC, so it may not be a bad thing to move this test there.

Reply:
> I think the current state is fine.

OK. Great.

> Note that `CompareException` is already imported in `Standard.Base` (`Statistics.enso`).

Removing the import in 3db825ccb4

> Incidentally, the `ObjectComparator` is imported in `Standard.Table`, so maybe the best option would be to move the `Comparator_Spec` to `Table_Tests` instead? 

Haven't we planned on forbidding to import class from another's library `polyglot/java` directory? In any case, this is a complex topic and we can leave it for later.

Reply:
Can't you just move all the imports necessary for tests in some *special*, *hidden* module in `Standard.Base`? Something like `Standard/Base/src/SpecialImports.enso` that will contain a comment with link to this discussion, and will be imported by Main?

Reply:
Imports moved to `Extra_Imports.enso` file in aaa5e9ec9e